### PR TITLE
Data_Engine: Quality of life for DomainTree

### PR DESCRIPTION
**Pull request by kThorsager**
### Issues addressed by this PR
Closes #

some support methods for DomainTree to alleviate working with it

### Test files

Not needed

### Changelog

- added tolerance to ItemsInRange as well

### CODE REVIEW

**Comment by kThorsager** [comment]
Hm, the branch has the latest commit but not the PR, unsure of how that happened. (pushed it during the github issues, but I can see the commit on Github 🤷‍♂️)

File: Spatial_Engine/Create/DomainTree.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates a spatial data tree from the data. Useful for spatial queries in large data sets.")]
+        [Input("elements", "The elements to store in the data tree.")]
+        [Input("treeDegree", "Degree of the tree. Determinse the number of children each node of the tree can have.")]
```
Comment (pawelbaran) [inline Spatial_Engine/Create/DomainTree.cs:44]:
```suggestion
        [Input("treeDegree", "Degree of the tree. Determines the number of children each node of the tree can have.")]
```

File: Spatial_Engine/Create/DomainTree.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        [Input("leafSize", "Determinse the number of siblings a leaf node can have.")]
```
Comment (pawelbaran) [inline Spatial_Engine/Create/DomainTree.cs:45]:
```suggestion
        [Input("leafSize", "Determines the number of siblings a leaf node can have.")]
```

File: Spatial_Engine/Create/DomainTree.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        [Input("sampleSize", "The number of items used to determine how to split the collection.")]
+        [Output("domainTree", "A spatial data tree containing all the provided elements in its leaves.")]
```
Comment (pawelbaran) [inline Spatial_Engine/Create/DomainTree.cs:47]:
```suggestion
        [Output("domainTree", "A spatial data tree containing all provided elements in its leaves.")]
```

File: Spatial_Engine/Spatial_Engine.csproj [hunk 4]
```diff
@@ -32,27 +32,30 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="BHoM">
-<HintPath>C:\ProgramData\BHoM\Assemblies\BHoM.dll</HintPath>
+      <HintPath>C:\ProgramData\BHoM\Assemblies\BHoM.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Data_oM">
+      <HintPath>..\..\..\ProgramData\BHoM\Assemblies\Data_oM.dll</HintPath>
```
Comment (pawelbaran) [inline Spatial_Engine/Spatial_Engine.csproj]: the path should not be relative?

File: Data_Engine/Create/DomainTreeLeaf.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Create a leaf node for a DomainTree.")]
+        [Input("data", "The data to store in the DomainTree leaf.")]
+        [Input("domainBox", "A DomainBox which encompasses the data.")]
```
Comment (pawelbaran) [inline Data_Engine/Create/DomainTreeLeaf.cs:43]:
```suggestion
        [Input("domainBox", "A DomainBox that encompasses the data.")]
```

File: Spatial_Engine/Spatial_Engine.csproj (hunk 4 above)
Comment (kThorsager) [reply Spatial_Engine/Spatial_Engine.csproj]: Yes, it's a new world, but should already been fixed

**Pull request merged**